onnxruntime/test/framework/cuda/fence_cuda_test.cc

```diff
@@ -62,28 +62,30 @@ static common::Status LoadInferenceSessionFromModel(InferenceSession& session, o
 
 CREATE_INITIALIZER_FUNC(float, TensorProto_DataType_FLOAT, add_float_data)
 CREATE_INITIALIZER_FUNC(int64_t, TensorProto_DataType_INT64, add_int64_data)
-// TO DO: Figure out a way to enable it again
-TEST(CUDAFenceTests, DISABLED_PartOnCPU) {
```

Comment:
>DISABLED_PartOnCPU [](start = 21, length = 18)

This test was disabled before when graph is always serialized before session initialize, and node can have providers specified. I remember with serialization to ModelProto, session initialize won't see the execution provider and thus everything runs on CPU or GPU, no mix and match.

Have you checked if the code path you added is hit in this test?

Reply:
Yes I have checked.